ffmpeg_tools/commands.py

```diff
@@ -348,17 +348,107 @@ def find_unsupported_data_streams(metadata):
     ]
 
 
-def find_unsupported_subtitle_streams(metadata):
+def find_unsupported_subtitle_streams(metadata, target_container):
+    if target_container is None:
+        # This is a corner case that won't happen if you explicitly select
+        # the target container instead of letting ffmpeg select one
+        # based on the file name and possibly other factors.
+        # It's best to always specify the container but ffmpeg-tools allows
+        # you not to do it for the sake of flexibility.
+        #
+        # If the target container is selected implicitly by ffmpeg, we
+        # can't really know which subtitle streams it supports.
+        # The safest bet is to say that all subtitle streams are supported.
+        # This way nothing gets stripped and ffmpeg either converts them on
+        # its own or fails, letting the user know that it can't process the video.
+        return []
```

Comment:
It's already forbidden by validations. You won't get here if you run them and `container` is `None` - which is the only situation when this special case could be triggered. It will happen only if you run it despite validations telling you not to.

It's just that the commands can work just fine with a wider range of parameters than the validations allow. For example in they allow you not to specify the container and rely on the name of the output file. I kept this case working after we switched to always specifying the container simply because it was very easy to do (we already had it working) and I wasn't 100% sure at the time that we won't ever revert to relying on names. ffmpeg-tools is a library and I think it's fine to provide a bit more generality than the client needs.

But yeah, in practice Golem (and probably your service too) does not rely on this feature and as we add more functionality it seems ever more superfluous because some of that functionality (like subtitle conversion) cannot work if we don't know the target container. I think that keeping it working by not assuming anything and just hoping that ffmpeg can convert the subtitles is consistent with letting it decide the container too and I'd leave it as is here.

After this point we should probably just remove any support for not specifying the container in commands but I didn't want to introduce such a change as a part of this pull request.

Reply:
So you say that if we execution will get to subtitles validation it should fail earlier if format was None?
Ok I missed it.

Let it be as is. But maybe throw exception in validate_unsupported_subtitle_streams if format is None and write comment nearby.
Because now this check relies on assumption, that other function behaves as expected. And if format validation will change, subtitles validation will change behavior implicite.

Reply:
OK. I'll add an exception and a comment.

I'd personally use an `assert` for something like this since this is something that's impossible to happen unless someone changes the code and introduces a bug - which should be caught in development.

Reply:
Done: 87d0fce636f3b6a9d359e45d2bcab449e3f4e00a